packages/compass-connections/src/stores/connections-store.ts

```diff
@@ -578,7 +552,13 @@ export function useConnections({
         connectionInfo
       );
     } catch (error) {
-      connectingConnectionAttempt.current = undefined;
+      if ((error as Error).message === CONNECTION_CANCELED_ERR) {
+        dispatch({
+          type: 'cancel-connection-attempt',
+        });
+        return;
+      }
+
       if (connectionInfo) {
         onConnectionFailed(connectionInfo, error as Error);
       }
```

Comment:
What exactly are we checking for here? How is it possible that we don't have connection info at that point?